Build failure: Found incompatible Python interpreter (2.7) and Python libraries (3.7)

I am building with latest master branch. [9fd82b1647461de153c8da5eaa606f93cb933e34]

Looks clang-native get pass but not non-native. Issue is with lldb.

Error log:
-- Found PythonLibs: /data/saininav/work/workspace/poky/build-opencl/tmp/work/corei7-64-poky-linux/clang/9.0.0-r0/recipe-sysroot-native/usr/lib/libpython3.7m.so (found version "3.7.4") 
CMake Error at /data/saininav/work/workspace/poky/build-opencl/tmp/work-shared/llvm-project-source-9.0.0-r0/git/lldb/cmake/modules/LLDBConfig.cmake:229 (message):
  Found incompatible Python interpreter (2.7) and Python libraries (3.7)
Call Stack (most recent call first):
  /data/saininav/work/workspace/poky/build-opencl/tmp/work-shared/llvm-project-source-9.0.0-r0/git/lldb/CMakeLists.txt:20 (include)

> Please cherry-pick #163 to meta-clang and https://patchwork.openembedded.org/patch/164648/ to oe-core and report back

Build still fails with above patches. Same error.

| -- Looking for os_signpost_interval_begin
| -- Looking for os_signpost_interval_begin - not found
| -- Found PythonInterp: /data/saininav/work/workspace/poky/build-opencl/tmp/hosttools/python (found version "2.7.12")
| -- Constructing LLVMBuild project information
| -- Found Git: /data/saininav/work/workspace/poky/build-opencl/tmp/hosttools/git (found version "2.7.4")
| -- Linker detection: GNU ld
| -- Targeting X86
| -- Looking for sys/resource.h
| -- Looking for sys/resource.h - found
| -- Clang version: 9.0.0
| -- Performing Test CXX_SUPPORTS_NO_NESTED_ANON_TYPES_FLAG
| -- Performing Test CXX_SUPPORTS_NO_NESTED_ANON_TYPES_FLAG - Success
| -- Looking for include file sys/inotify.h
| -- Looking for include file sys/inotify.h - found
| -- LLD version: 9.0.0
| -- Found LibEdit: /data/saininav/work/workspace/poky/build-opencl/tmp/work/corei7-64-poky-linux/clang/9.0.0-r0/recipe-sysroot-native/usr/include (found version ".")
| -- Looking for el_winsertstr
| -- Looking for el_winsertstr - found
| -- Looking for stdint.h
| -- Looking for stdint.h - found
| -- Looking for stddef.h
| -- Looking for stddef.h - found
| -- Check size of el_rfunc_t
| -- Check size of el_rfunc_t - done
| -- Found PythonLibs: /data/saininav/work/workspace/poky/build-opencl/tmp/work/corei7-64-poky-linux/clang/9.0.0-r0/recipe-sysroot-native/usr/lib/libpython3.7m.so (found version "3.7.4")
| CMake Error at /data/saininav/work/workspace/poky/build-opencl/tmp/work-shared/llvm-project-source-9.0.0-r0/git/lldb/cmake/modules/LLDBConfig.cmake:229 (message):
|   Found incompatible Python interpreter (2.7) and Python libraries (3.7)
| Call Stack (most recent call first):
|   /data/saininav/work/workspace/poky/build-opencl/tmp/work-shared/llvm-project-source-9.0.0-r0/git/lldb/CMakeLists.txt:20 (include)

--- a/recipes-devtools/clang/clang_git.bb
+++ b/recipes-devtools/clang/clang_git.bb
@@ -134,6 +134,7 @@ EXTRA_OECMAKE_append_class-nativesdk = "\
                   -DLLDB_TABLEGEN=${STAGING_BINDIR_NATIVE}/lldb-tblgen \
                   -DPYTHON_LIBRARY=${STAGING_LIBDIR}/lib${PYTHON_DIR}${PYTHON_ABI}.so \
                   -DPYTHON_INCLUDE_DIR=${STAGING_INCDIR}/${PYTHON_DIR}${PYTHON_ABI} \
+                  -DPYTHON_EXECUTABLE='${PYTHON}' \
 "
 EXTRA_OECMAKE_append_class-target = "\
                   -DCMAKE_CROSSCOMPILING:BOOL=ON \

